### PR TITLE
Add proper licences for activated chapters

### LINKED ISSUE #910: Proper licencing needs to be added for activated chapters
Now that the licensing structure for the Field Guide project has been finalised in #863, we need to retroactively update licencing for all changes made previously.

This is a subset of #451: It is specifically about content already refactored/activated and merged.

- Base content (Bootstrap, Jquery etc)
- FSA module
- Chapters
  - Algorithms
  - HCI
  - Data Representation
  - Coding - Introduction
  - Coding - Compression
  - Coding - Encryption
  - Coding - Error Control

### PR DESCRIPTION
**Pull request by eAlasdair**
Resolves #910 
Related to #451

Note that in most cases this *does not* update interactive READMEs, as there are unrelated issues with them also.

Reviewers, please pay close attention to items in the list with associated notes - their licences deviated from the norm in some way and I could have improperly licensed them as a result.

Added Licences for:
- [x] Base Dependencies
  - [x] Bootstrap
  - [x] jQuery
  - [x] Details Element Polyfill
  - [x] Featherlight
  - [x] Popper.js
  - [x] Multiple Select
  - [x] StickyState (*no LICENSE found, assumed MIT from npm* - [issue here](https://github.com/soenkekluth/sticky-state/issues/32))
  - [x] iframe resizer
- [x] FSA module
  - [x] es6 ArrayMap
  - [x] Underscore
- [x] Development Dependencies
  - [x] autoprefixer
  - [x] @babel/core
  - [x] @babel/preset-env
  - [x] browserify
  - *removed child_process from the project (issue with source [here](https://www.npmjs.com/package/child_process))*
  - [x] del
  - [x] gulp
  - [x] gulplog
  - [x] gulp-babel
  - [x] gulp-filter
  - [x] gulp-if
  - [x] gulp-jshint
  - [x] gulp-error-handle (*no source found beyond npm, MIT licence assumed from npm* - [see here](https://www.npmjs.com/package/gulp-error-handle))
  - [x] gulp-notify
  - [x] gulp-postcss
  - [x] gulp-rename
  - [x] gulp-sass
  - [x] gulp-sourcemaps
  - [x] gulp-tap
  - [x] gulp-terser
  - [x] gulp-util (*deprecated* [see here](https://www.npmjs.com/package/gulp-util))
  - [x] jshint (*JSON Licence mentioned [in npm](https://www.npmjs.com/package/jshint) but only MIT stated in source*)
  - [x] jshint-stylish
  - [x] puppeteer
  - [x] request
  - [x] node-gyp
  - [x] postcss-flexbugs-fixes
  - [x] run-sequence
  - [x] vinyl-buffer
  - [x] yargs
- [x] Libraries required for specific interactives
  - Algorithms chapter
    - Sorting Algorithms
      - [x] Dragula
  - Data Representation chapter
    - Unicode Binary
      - [x] urijs
      - [x] underscore.string
      - [x] string.fromcodepoint
    - Unicode Chars
      - [x] string.fromcodepoint
    - CMY Mixer
      - [x] nouislider
      - [x] wnumb
    - RGB Mixer
      - [x] nouislider
      - [x] wnumb
    - Colour Matcher
      - [x] nouislider
      - [x] wnumb
  - Coding Compression chapter
    - Huffman Tree
      - [x] viz-lite (*source file*)
    - JPEG Compression
      - [x] interactjs (*alpha*)
      - [x] dct
  - Coding Encryption chapter
    - Big number Calculator
      - [x] big.js
    - Caesar Cipher
      - [x] es5-shim
      - [x] es6-shim
    - Frequency Analysis
      - [x] chart.js
      - [x] es5-shim
      - [x] es6-shim
    - RSA no padding
      - [x] jsencrypt
    - SHA2
      - [x] crypto-js
  - Unspecified Chapter
    - RSA JSencrypt
      - [x] jsencrypt
- [x] requirements/base
  - [x] wheel
  - [x] Pillow
  - [x] django-inline-svg
  - [x] selenium
  - [x] Verto
  - [x] python-markdown-math
  - [x] django-haystack (*License file not found - copyright holder stated on their [website](https://haystacksearch.org/)*)
  - [x] Whoosh
  - [x] django-widget-tweaks
  - [x] lxml
  - [x] cssselect
  - [x] PyYAML
  - [x] django-modeltranslation
  - [x] uniseg
  - [x] python-bidi
  - [x] django-bidi-utils
  - [x] django-statici18n
- [x] requirements/local
  - [x] sphinx
  - [x] sphinx-rtd-theme (*several license files included [here](https://github.com/rtfd/sphinx_rtd_theme/), only MIT listed*)
  - [x] django-debug-toolbar
- [x] requirements/test
  - [x] flake8
  - [x] pydocstyle
  - [x] coverage
  - [x] django-test-without-migrations
- [x] other
  - [x] Django (*includes 'code from the Python standard library'*)

### CODE REVIEW

**Comment by JackMorganNZ** [comment]
@eAlasdair For `child_process`, the only place it is used is to make a beep sound on a error, which I don't think we've ever implemented. We don't need a audio error sound, visual text is fine, so easiest solution is to remove it:

- `csfieldguide/package.json` - Remove line 42.
- `csfieldguide/gulpfile.js` - Remove line 21, 54 to 65, and 68.

File: LICENCE-THIRD-PARTY [hunk 1]
```diff
@@ -41,6 +159,510 @@ licensed under MIT licence.
...
+third-party-licences/django-debug-toolbar.txt
+==============================================================================
+
+==============================================================================
+django-haystack (AKA Haystack)
+------------------------------------------------------------------------------
+http://haystacksearch.org/
+Copyright 2009-2011 Daniel Lindsley
+licensed under BSD licence.
+No known licence file to include
```
Comment (courtneycb) [inline LICENCE-THIRD-PARTY]: Is this relevant? https://github.com/django-haystack/django-haystack/blob/master/LICENSE

File: LICENCE-THIRD-PARTY [hunk 2]
```diff
@@ -57,4 +679,104 @@ https://github.com/tweenjs/tween.js/
...
+third-party-licences/wheel.txt
+==============================================================================
+
+==============================================================================
+Whoosh
+------------------------------------------------------------------------------
+https://bitbucket.org/mchaput/whoosh/overview
+Copyright 2011 Matt Chaput
+licensed under BSD licence.
+third-party-licences/wheel.txt
```
Comment (courtneycb) [inline LICENCE-THIRD-PARTY]: Incorrect link. Change wheel to whoosh

File: LICENCE-THIRD-PARTY [hunk 3]
```diff
@@ -57,4 +679,104 @@ https://github.com/tweenjs/tween.js/
...
+==============================================================================
+
+==============================================================================
+wnumb
+------------------------------------------------------------------------------
+https://github.com/leongersen/wnumb
+Created by Leon Gersen
+licensed under WTFPL
```
Comment (courtneycb) [inline LICENCE-THIRD-PARTY]: Should we add 'No known licence file to include' here like you have elsewhere?

File: LICENCE-THIRD-PARTY (hunk 1 above)
Comment (eAlasdair) [reply LICENCE-THIRD-PARTY]: Yes, how did I miss that??

File: LICENCE-THIRD-PARTY (hunk 3 above)
Comment (eAlasdair) [reply LICENCE-THIRD-PARTY]: WTFPL doesn't really have a licence per se
http://www.wtfpl.net/
Should I do it anyway?

**Pull request merged**